intake/container/__init__.py

```diff
@@ -0,0 +1,16 @@
+from .base import BaseContainer
+from .dataframe import DataFrame
+from .ndarray import NdArray
+from .semistructured import SemiStructured
+
+container_map = {
+    'dataframe' : DataFrame,
+    'python' : SemiStructured,
+    'ndarray' : NdArray
+}
+
+def get_container_klass(container):
+    return container_map.get(container, BaseContainer)
```

Comment:
Totally fine with that. Later we may provide a "register" API, function or decorator.